### PR TITLE
python312Packages.pytorch-pfn-extras: 0.7.6 -> 0.7.7

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Fix and update the package.

Changelog: https://github.com/pfnet/pytorch-pfn-extras/releases/tag/v0.7.7

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 342135` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python311Packages.ffcv</li>
    <li>python311Packages.ffcv.dist</li>
    <li>python311Packages.pytorch-pfn-extras</li>
    <li>python311Packages.pytorch-pfn-extras.dist</li>
    <li>python312Packages.pytorch-pfn-extras</li>
    <li>python312Packages.pytorch-pfn-extras.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 342135` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python311Packages.ffcv</li>
    <li>python311Packages.ffcv.dist</li>
    <li>python311Packages.pytorch-pfn-extras</li>
    <li>python311Packages.pytorch-pfn-extras.dist</li>
    <li>python312Packages.pytorch-pfn-extras</li>
    <li>python312Packages.pytorch-pfn-extras.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 342135` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.ffcv</li>
    <li>python311Packages.ffcv.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pytorch-pfn-extras</li>
    <li>python311Packages.pytorch-pfn-extras.dist</li>
    <li>python312Packages.pytorch-pfn-extras</li>
    <li>python312Packages.pytorch-pfn-extras.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 342135` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.ffcv</li>
    <li>python311Packages.ffcv.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pytorch-pfn-extras</li>
    <li>python311Packages.pytorch-pfn-extras.dist</li>
    <li>python312Packages.pytorch-pfn-extras</li>
    <li>python312Packages.pytorch-pfn-extras.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 342135` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.ffcv</li>
    <li>python311Packages.ffcv.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pytorch-pfn-extras</li>
    <li>python311Packages.pytorch-pfn-extras.dist</li>
    <li>python312Packages.pytorch-pfn-extras</li>
    <li>python312Packages.pytorch-pfn-extras.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 342135` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.ffcv</li>
    <li>python311Packages.ffcv.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pytorch-pfn-extras</li>
    <li>python311Packages.pytorch-pfn-extras.dist</li>
    <li>python312Packages.pytorch-pfn-extras</li>
    <li>python312Packages.pytorch-pfn-extras.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/pytorch-pfn-extras/default.nix [hunk 1]
```diff
@@ -1,39 +1,47 @@
...
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "pfnet";
     repo = "pytorch-pfn-extras";
     rev = "refs/tags/v${version}";
-    hash = "sha256-vSon/0GxQfaRtSPsQbYAvE3s/F0HEN59VpzE3w1PnVE=";
+    hash = "sha256-0+ltkm7OH18hlpHYyZCmy1rRleF52IM2BjLoW44tJUY=";
   };
+  # src = /home/gaetan/pytorch-pfn-extras;
```
Comment (pbsds) [inline pkgs/development/python-modules/pytorch-pfn-extras/default.nix:36]:
```suggestion
```

File: pkgs/development/python-modules/pytorch-pfn-extras/default.nix [hunk 2]
```diff
@@ -1,39 +1,47 @@
...
-      hash = "sha256-85UDGcgJyQS5gINbgpNM58b3XJGvf+ArtGhwJ5EXdhk=";
-    })
-  ];
+  # patches = [
+  #   (fetchpatch {
+  #     name = "relax-setuptools.patch";
+  #     url = "https://github.com/pfnet/pytorch-pfn-extras/commit/96abe38c4baa6a144d604bdd4744c55627e55440.patch";
+  #     hash = "sha256-85UDGcgJyQS5gINbgpNM58b3XJGvf+ArtGhwJ5EXdhk=";
+  #   })
+  # ];
```
Comment (pbsds) [inline pkgs/development/python-modules/pytorch-pfn-extras/default.nix:44]:
```suggestion
```

File: pkgs/development/python-modules/pytorch-pfn-extras/default.nix (hunk 1 above)
Comment (GaetanLepage) [reply pkgs/development/python-modules/pytorch-pfn-extras/default.nix:36]: Oops.

File: pkgs/development/python-modules/pytorch-pfn-extras/default.nix [hunk 3]
```diff
@@ -54,10 +52,24 @@ buildPythonPackage rec {
...
   ];
 
   pythonImportsCheck = [ "pytorch_pfn_extras" ];
 
+  disabledTests = [
+    # AssertionError: assert 4 == 0
+    # where 4 = <MagicMock id='140733587469184'>.call_count
+    "test_lr_scheduler_wait_for_first_optimizer_step"
+
+    # torch.distributed is not available on darwin
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pytorch-pfn-extras/default.nix:67]: Shouldn't this then be darwin specific?

Comment (GaetanLepage) [reply pkgs/development/python-modules/pytorch-pfn-extras/default.nix:67]: It should indeed. Fixed.

Comment (GaetanLepage) [reply pkgs/development/python-modules/pytorch-pfn-extras/default.nix]: The `test_lr_scheduler_wait_for_first_optimizer_step` should be skipped on all platforms though.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 342135` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python311Packages.ffcv</li>
    <li>python311Packages.ffcv.dist</li>
    <li>python311Packages.pytorch-pfn-extras</li>
    <li>python311Packages.pytorch-pfn-extras.dist</li>
    <li>python312Packages.pytorch-pfn-extras</li>
    <li>python312Packages.pytorch-pfn-extras.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 342135` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python311Packages.ffcv</li>
    <li>python311Packages.ffcv.dist</li>
    <li>python311Packages.pytorch-pfn-extras</li>
    <li>python311Packages.pytorch-pfn-extras.dist</li>
    <li>python312Packages.pytorch-pfn-extras</li>
    <li>python312Packages.pytorch-pfn-extras.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 342135` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.ffcv</li>
    <li>python311Packages.ffcv.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pytorch-pfn-extras</li>
    <li>python311Packages.pytorch-pfn-extras.dist</li>
    <li>python312Packages.pytorch-pfn-extras</li>
    <li>python312Packages.pytorch-pfn-extras.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 342135` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.ffcv</li>
    <li>python311Packages.ffcv.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pytorch-pfn-extras</li>
    <li>python311Packages.pytorch-pfn-extras.dist</li>
    <li>python312Packages.pytorch-pfn-extras</li>
    <li>python312Packages.pytorch-pfn-extras.dist</li>
  </ul>
</details>

**Review by pbsds — APPROVED** [review]
lgtm, assuming eval passes

**Pull request merged**